src/rlic/_boundaries.py

```diff
@@ -0,0 +1,110 @@
+import sys
+from typing import Final, NewType, TypeAlias, TypedDict
+
+if sys.version_info < (3, 11):
+    from exceptiongroup import ExceptionGroup
+
+Bound = NewType("Bound", str)
+BoundPair = tuple[Bound, Bound]
+AnyBound: TypeAlias = Bound | BoundPair
+
+
+class AbstractBounds(TypedDict):
+    x: AnyBound
+    y: AnyBound
+
+
+class ConcreteBounds(TypedDict):
+    x: BoundPair
+    y: BoundPair
+
+
+def as_pair(b: AnyBound) -> BoundPair:
+    match b:
+        case (str(), str()):
```

Comment:
I should reorder branches from simplest to most complex, for a consistent logical flow.